tests/sdk.test.ts

```diff
@@ -104,6 +138,19 @@ describe("SDK tests", () => {
       method: "POST",
     });
     expect(res.status).toEqual(500);
+    expect(createComment).toHaveBeenCalledWith({
+      issue_number: 5,
+      owner: "ubiquibot",
+      repo: "bot",
+      body: `\`\`\`diff
+! test error
+\`\`\`
+<!--
+{
+  "caller": "error"
+}
+-->`,
+    });
```

Comment:
I'm not sure if logger is working correctly as there is not stack trace? @gentlementlegen or @Keyrxng

Reply:
From one of my QA comments:

```
\`\`\`diff
! keyrxng2 you were previously unassigned from this task. You cannot reassign yourself.
\`\`\`
<!--
{
  "error": {},
  "caller": "startStopTask"
}
-->
```

We do get a full error stack trace in the worker/action logs... `caller:` is the function which the error happened in right so it would be whatever fn name is for caller and then whatever metadata object gets passed to the logger should also be commented. But we should also have a full stack trace being commented as well?

Reply:
Hmm it would be nice to have stack trace in the comment but thinking about it more it's probably not good for security. 
However caller being `error` is incorrect, is it maybe because the logger is called in anonymous function?

Reply:
I think because we might be losing the original stack trace as the logger recreates the stack internally I'm sure... would need looked into but typically we do see from where the call failed exactly with a code snippet